### PR TITLE
[IOTDB-2725] MPP query engine interface definition

### PR DESCRIPTION
**Pull request by JackieTien97**
In this pr, we define:

1. all the PlanNode which assembles the logic plan tree in package `org.apache.iotdb.db.mpp.plan.node`
2. Interface of DataBlockManager which we need to send and receive TsBlock from remote node, in package `org.apache.iotdb.db.mpp.buffer`
3. some common data structure like `QueryId`, in package `org.apache.iotdb.db.mpp.common`
4. query execution related things like query scheduler, `ExecFragmentInstance` and so on, in package `org.apache.iotdb.db.mpp.execution`
5. `Operator` is an interface for all runnable operators and we have not implemented all the operators we listed in [Confluence](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=199538684), any concrete operator should implement `Operator` interface, in package `org.apache.iotdb.db.mpp.operator`

### CODE REVIEW

**Review by ericpai — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/buffer/SinkHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.buffer;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.nio.ByteBuffer;
+
+public interface SinkHandler {
```
Comment (cornmonster) [inline server/src/main/java/org/apache/iotdb/db/mpp/buffer/SinkHandler.java:25]: 这个命名为SinkHandle吧，表示的是类似FileHandle的语义，和handler不太一样

File: server/src/main/java/org/apache/iotdb/db/mpp/buffer/SourceHandler.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.buffer;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.io.Closeable;
+import java.nio.ByteBuffer;
+
+public interface SourceHandler extends Closeable {
```
Comment (cornmonster) [inline server/src/main/java/org/apache/iotdb/db/mpp/buffer/SourceHandler.java:26]: SourceHandle

File: server/src/main/java/org/apache/iotdb/db/mpp/buffer/SinkHandler.java (hunk 1 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/mpp/buffer/SinkHandler.java:25]: ok

File: server/src/main/java/org/apache/iotdb/db/mpp/buffer/SourceHandler.java (hunk 2 above)
Comment (JackieTien97) [reply server/src/main/java/org/apache/iotdb/db/mpp/buffer/SourceHandler.java:26]: ok

**Comment by JackieTien97** [comment]
@cornmonster PTAL : )

**Review by xingtanzjr — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/node/process/WithoutNode.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.plan.node.process;
+
+import org.apache.iotdb.db.mpp.common.WithoutPolicy;
+import org.apache.iotdb.db.mpp.plan.node.PlanNodeId;
+
+/** WithoutNode is used to discard specific rows from upstream node. */
+public class WithoutNode extends ProcessNode {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/node/process/WithoutNode.java:25]:
```suggestion
public class FilterNullNode extends ProcessNode {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  // Analyze the statement in QueryContext. Generate the analysis this query need
+  public void analyze() {
+    // initialize the variable `analysis`
+
+  }
+
+  // Use LogicalPlanner to do the logical query plan and logical optimization
+  public void doLogicalPlan() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java:71]:
```suggestion
  public void generateLogicalPlan() {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    LogicalPlanner planner = new LogicalPlanner(this.analysis, this.context, this.planOptimizers);
+    this.logicalPlan = planner.plan();
+  }
+
+  // Generate the distributed plan and split it into fragments
+  public void doDistributedPlan() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java:77]:
```suggestion
  public void generateDistributedPlan() {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * QueryExecution stores all the status of a query which is being prepared or running inside the MPP
+ * frame. It takes three main responsibilities: 1. Prepare a query. Transform a query from statement
+ * to DistributedQueryPlan with fragment instances. 2. Dispatch all the fragment instances to
+ * corresponding physical nodes. 3. Collect and monitor the progress/states of this query.
+ */
+public class QueryExecution {
+  private QueryContext context;
+  private QueryScheduler scheduler;
+  private QueryStateMachine stateMachine;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/QueryExecution.java:38]: Use the stateMachine in QueryScheduler?

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**